tokio/src/task/coop/mod.rs

```diff
@@ -413,7 +499,7 @@ cfg_coop! {
     /// If the future exceeds its budget while being polled, control is yielded back to the
     /// runtime.
     #[inline]
-    pub(crate) fn cooperative<F: Future>(fut: F) -> Coop<F> {
+    pub fn cooperative<F: Future>(fut: F) -> Coop<F> {
         Coop { fut }
```

Comment:
Could we have an example or something. This is really short.

It could worth to add something like "this should be used to make futures from external crates cooperative". And you shouldn't use it with Tokio futures that are already cooperative. Maybe an example with the mpsc channel from futures and making `recv` cooperative.